sw/device/silicon_creator/lib/boot_svc/boot_svc_header.h

```diff
@@ -52,14 +61,21 @@ OT_ASSERT_MEMBER_OFFSET(boot_svc_header_t, type, 36);
 OT_ASSERT_MEMBER_OFFSET(boot_svc_header_t, length, 40);
 OT_ASSERT_SIZE(boot_svc_header_t, CHIP_BOOT_SVC_MSG_HEADER_SIZE);
 
-enum {
-  /**
-   * Common identifier shared by all boot services messages.
-   *
-   * ASCII "BSVC".
-   */
-  kBootSvcIdentifier = 0x43565342,
-};
+/**
+ * Initialize the header of a boot services message.
+ *
+ * This function assumes that message payload starts immediately after the
+ * header and is exactly `length - sizeof(boot_svc_header_t)` bytes for digest
+ * computation. Since this function also intializes the message digest as part
+ * of header initialization, it must be called after the message payload is
+ * initialized.
+ *
+ * @param type Message type.
+ * @param length Total length of the message in bytes.
+ * @param[out] header Output buffer for the message.
+ */
+void boot_svc_header_init(uint32_t type, uint32_t length,
```

Comment:
nit: Since this is called after the message is populated, should it be called something like `boot_svc_header_build` or `*_calc`? This is so that the function name does not lend itself to be used before the message has been populated.

Reply:
Good point, I agree and thanks for the suggestions. I thiknk I'll go with `boot_svc_header_finalize()` to make it clear that the function is the last one to be called when initializing a message